pkg/image/registry/imagestream/etcd/etcd.go

```diff
@@ -91,6 +91,17 @@ func (r *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, boo
 	return r.store.Update(ctx, obj)
 }
 
+// UpdateAll updates both the image stream specification and status.
+func (r *REST) UpdateAll(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	oldStrategy := r.store.UpdateStrategy
+	r.store.UpdateStrategy = imagestream.NewAllStrategy(oldStrategy.(imagestream.Strategy))
+	defer func() {
+		r.store.UpdateStrategy = oldStrategy
```

Comment:
https://github.com/openshift/origin/pull/3982#discussion_r42655808 is the reason for re-review.